qermit/zero_noise_extrapolation/zne.py

```diff
@@ -82,7 +82,11 @@ def circuit(circ: Circuit, noise_scaling: int, **kwargs) -> Circuit:
             folded_circ.add_barrier(folded_circ.qubits + folded_circ.bits)
 
             # Add original circuit
-            folded_circ.add_circuit(circ, folded_circ.qubits, folded_circ.bits)
+            for gate in circ.get_commands():
+                if gate.op.type == OpType.Barrier:
+                    folded_circ.add_barrier(gate.args)
```

Comment:
this may require more conditions - I don't think a CircBox `gate` can be added with `add_gate` but would need to be added via `add_circbox`, or should every gate in the circuit either be a  Barrier or can be added via `add_gate`? If the latter, and if its a smaller set of gates, it might be worth adding ifs for their types, and then else throwing some error.